pageserver/src/tenant/delete.rs

```diff
@@ -478,7 +477,7 @@ impl DeleteTenantFlow {
                 Ok(())
             }
             .instrument({
-                let span = tracing::info_span!(parent: None, "delete_tenant", tenant_id=%tenant_id);
+                let span = tracing::info_span!(parent: None, "delete_tenant", tenant_id=%tenant_shard_id.tenant_id, shard_id=%tenant_shard_id.shard_slug());
```

Comment:
This is quite nauseating that we now need to carry three.

Need to start using Valuable ... but that is behind an unstable flag. Ok, so carrying three.

In the meantime perhaps something like this could work instead?

```
// tenant_id should just be a &TenantId, shardid should just be a copy
let (tenant_id, shard_id) = tenant_shard_id.split();
let span = tracing::info_span!(parent: None, "delete_tenant", %tenant_id, %shard_id);
```

Reply:
That helps a bit when doing info_span!, although there are many cases of #[instrument] where one doesn't have the opportunity to call a helper.  I feel a bit ambivalent about adding such a helper: it makes the span line nicer, but it also polllutes the calling function with local tenant_id, shard_id variables: there is some virtue in confining the visual noise to the span line, so that it's obvious to readers that picking out the tenant_id and shard_id is just for observability and nothing else.

Our end state should probably be to have a Valuable TenantShardId _and_ a Valuable TenantShardTimelineId type, so that either tenant-scope or timeline-scope things can just pass a single identity object into span macros.  Until then, our slightly ugly spans with explicit tenant and timeline will be slightly-more-ugly with the addition of shard ID (but it's the right thing to do, to prioritize the quality of the log output by having separate labels).